sqlglot/dialects/teradata.py

```diff
@@ -105,6 +106,19 @@ def _parse_update(self) -> exp.Expression:
                 },
             )
 
+        def _parse_rangen(self):
+            this = self._parse_id_var()
+            self._match(TokenType.BETWEEN)
+
+            expressions = []
+            while self._next and not self._match_text_seq("EACH"):
+                self._match(TokenType.COMMA)
+                expressions.append(self._parse_conjunction() or self._parse_expression())
```

Comment:
> EACH allows date intervals like "INTERVAL '1' MONTH", but since there's no alias allowed _parse_conjunction should work. I'll update that and the expressions.

Maybe even `_parse_bitwise` is just fine, although I don't remember the exact call sequence. I always look it up if needed.

> I assume I should choose the parser as far down the list as possible?

That's correct. In that way, you reduce function call overhead and possible errors that might result from "overshooting" with regard to what you're able to parse at a given step. If you just want an id var, it'd be a waste to call `_parse_expression`, because that'd lead to many unnecessary function calls that will probably lead to `_parse_id_var` anyway.